ref/general/zero-migration-architecture.adoc

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2020 IBM Corporation and others.
+// Licensed under Creative Commons Attribution-NoDerivatives
+// 4.0 International (CC BY-ND 4.0)
+//   https://creativecommons.org/licenses/by-nd/4.0/
+//
+// Contributors:
+//     IBM Corporation
+//
+:page-description: With zero-migration architecture, you can move to the latest version of Open Liberty with minimal impact to your current applications and configurations.
+:seo-title: Zero-migration architecture
+:seo-description: With zero-migration architecture, you can move to the latest version of Open Liberty with minimal impact to your current applications and configurations.
+:page-layout: general-reference
+:page-type: general
+= Zero migration architecture
+
+Zero migration architecture is a core design principle of Open Liberty, which supports pluggable features and full compatibility between product versions.
+With zero-migration architecture, you can move to the latest version of Open Liberty with minimal impact to your current applications and configurations.
+
+Zero-migration architecture means that you can use your existing, unmodified configuration and application files with an updated version of Open Liberty without unexpected changes in application behavior.
+You can update Open Liberty without updating your configuration files.
+Your existing APIs and behaviors are supported in new product versions, and new APIs and behaviors are added in new features and feature versions.
+
+== User configuration files
+
+Open Liberty never modifies user configuration files, which are fully compatible between versions.
+You can use a single version of your configuration files across multiple product versions.
+You can use files that you created for a previous version of Open Liberty with a later version.
+You can use files that you create for later versions with previous versions.
+As a result, if all configured features are installed, you can use a single set of configuration files across multiple versions without modifications.
+Open Liberty ignores any configuration settings that do not apply to the running version of the runtime environment.
+User configuration files do not require modification when the runtime is updated, either for a fix pack or a new version.
+
+Open Liberty provides an absolute separation between product files and user files.
+Product files do not need to be modified by users.
+By default, user files such as server configuration and application files are located under a separate directory from the product files.
+If you want to override the default location, for example to put them on a separate file system from the product files you can set the `WLP_USER_DIR` environment variable.
```

Comment:
Maybe rework this sentence so it's clearer. I'm not sure I understand what "them" is referring to: 
"If you want to override the default location, for example, to put them on a separate file system from the product files, you can set the `WLP_USER_DIR` environment variable.